packages/babel-plugin-transform-es2015-parameters/src/default.js

```diff
@@ -44,17 +50,33 @@ const iifeVisitor = {
 };
 
 export const visitor = {
-  Function(path) {
+  Function(path, state) {
     const { node, scope } = path;
     if (!hasDefaults(node)) return;
 
     // ensure it's a block, useful for arrow functions
     path.ensureBlock();
 
-    const state = {
-      iife: false,
-      scope: scope,
-    };
+    const params = path.get("params");
+
+    if (state.opts.loose) {
+      const body = [];
+      for (let i = 0; i < params.length; ++i) {
+        const param = params[i];
+        if (param.isAssignmentPattern()) {
+          body.push(buildLooseDefaultParam({
+            VARIABLE_NAME: param.get("left"),
```

Comment:
We can go that route, but I think there's an error in the code. Should be:

```js
  let {a: { b : c} = {}} = _temp === undefined ? {} : _temp;
```

Otherwise, we would have to do the defaults before the destructuring:

```js
if (a === undefined) {
  a = {};
}
let { a: { b: c } = {} } = a;